support/routes/v2/isobot/index.ts

```diff
@@ -1,15 +1,28 @@
-import express, { RequestHandler } from "express"
+import { App, ExpressReceiver } from "@slack/bolt"
+import { RequestHandler } from "express"
 
 import { Whitelist } from "@database/models"
+import config from "@root/config/config"
 import logger from "@root/logger/logger"
 import WhitelistService from "@root/services/identity/WhitelistService"
 
 import BotService, { SlackPayload } from "./ops/botService"
 
-export const isobotRouter = express.Router()
 const botService = new BotService(
   new WhitelistService({ repository: Whitelist })
 )
+const signingSecret = config.get("slackbot.secret")
+const token = config.get("slackbot.token")
+
+const botReceiver = new ExpressReceiver({ signingSecret, endpoints: "/" })
+export const isobotRouter = botReceiver.router
+
+const bot = new App({
```

Comment:
i'm not very sure here. in this case, we want bolt to act as a router, not as a server. this means that we want bolt to capture a subset of routes and respond to those with a registered handler rather than capturing all incoming network data over the specified port (what `.start` does i think)

again, i haven't tested/deployed this so i can't be sure but i think this is my understanding

Reply:
tested with a `echo` command (not committed) against a ngrok forwarding to my local instance. this worked but with a huge caveat - we omit using middleware for the bot. this is because bolt runs their own middleware (see below ss) which causes forwarding from express -> bolt to fail 

i'm assuming this is ok because bolt _should_ enforce the security measures on their end via their inbuilt middleware. an alternative is we could manually check the middlewares until we find the one that causes the failure (not worth imo). an exhaustive list of the middleware: 

```ts

  if (isSecure) {
    // Our server only receives requests from the alb reverse proxy, so we need to use the client IP provided in X-Forwarded-For
    // This is trusted because our security groups block all other access to the server
    app.set("trust proxy", true)
  }
  app.use(helmet())

  // use growthbook across routes
  app.use(featureFlagMiddleware)

  app.use(
    cors({
      origin: FRONTEND_URL,
      credentials: true,
    })
  )
  app.use(express.json({ limit: "7mb" }))
  app.use(express.urlencoded({ extended: false }))
  app.use(cookieParser())
  app.use(express.static(path.join(__dirname, "public")))
  app.use(nocache())
}
```

<img width="433" alt="image" src="https://github.com/isomerpages/isomercms-backend/assets/44049504/17d29a40-a10d-4233-8967-6ca45ee2e46e">

![image](https://github.com/isomerpages/isomercms-backend/assets/44049504/cb27992b-b54e-4874-9a98-831225c3b842)

Reply:
to close up the loop on this, we are using express as the server and bolt as the router? bolt has it's own internal (unexposed) middlewares which does things like hmac verification etc.?

Reply:
yes, check the source [here](https://github.com/slackapi/bolt-js/blob/ddeb601b0d140b47e93e08305c23ad2cef4508b9/src/receivers/ExpressReceiver.ts#L430)